docs/AVS-Guide.md

```diff
@@ -0,0 +1,85 @@
+# Purpose
+This document aims to describe and summarize how actively validated services (AVSs) building on EigenLayer interact with the core EigenLayer protocol. Currently, this doc explains how AVS developers can use the APIs for: 
+- enabling operators to opt-in to the AVS,
+- enabling operators to withdraw stake from AVS, and
+- enabling AVS to freeze operators for the purpose of slashing (the corresponding unfreeze actions are determined by the veto committee).
+
+We are currently in the process of implementing the API for payment flow from AVSs to operators in EigenLayer. Details of this API will be added to this document in the near future.
+
+The following figure summarizes scope of this document: 
+![Doc Outline](./images/middleware_outline_doc.png)
+
+# Introduction
+In designing EigenLayer, the EigenLabs team aspired to make minimal assumptions about the structure of AVSs built on top of it. If you are getting started looking at the EigenLayer codebase, the `Slasher.sol` contains most of the logic that actually mediates the interactions between EigenLayer and AVSs. Additionally, there is a general-purpose [/middleware/ folder](https://github.com/Layr-Labs/eigenlayer-contracts/tree/master/src/contracts/middleware), which contains code that can be extended, used directly, or consulted as a reference in building an AVS on top of EigenLayer.
+
+## Important Terminology
+- **Tasks** - A task in EigenLayer is the smallest unit of work that operators commit to perform when serving an AVS. These tasks may be associated with one or more slashing conditions applicable to the AVS. 
+- **Strategies** - A strategy in EigenLayer is a contract that holds staker deposits, i.e. it controls one or more restaked asset(s). At launch EigenLayer will feature only simple strategies which may hold a single token. However, EigenLayer's strategy design is flexible and open, and in the future strategies could be deployed which implement more complex logic, including DeFi integrations. 
+- **Quorums** - A quorum in EigenLayer is a grouping of specific kinds of stake who opt into an AVS while satisfying a particular trait. Examples of such trait could be stETH stakers or native stakers.  The prupose of having a quorum is that a AVS can customize the makeup of their security offering by choosing which kinds of stake/security they would like to utilize.  
```

Comment:
typo: 'prupose' => 'purpose'
I think I'd contrast "LST restakers" with "native restakers" rather than "stETH stakers or native stakers". not too sure the terminology "trait" applies well here, but I'm OK with it